### PR TITLE
Handles overrides in a cleaner way

### LINKED ISSUE #307: Input presence checks don't account for overrides
```python
from hamilton import base, driver, ad_hoc_utils
def foo(baz: int) -> int:
    return baz + 1

def bar(foo: int) -> int:
    return foo + 1
    
mod = ad_hoc_utils.create_temporary_module(foo, bar)
dr = driver.Driver({}, mod, adapter=base.DefaultAdapter())
dr.execute(['bar'], inputs={'baz' : 10})['bar'] # 12 = 10 + 1 + 1

mod = ad_hoc_utils.create_temporary_module(foo, bar)
dr = driver.Driver({}, mod, adapter=base.DefaultAdapter())
dr.execute(['bar'], overrides={'foo' : 10}, inputs={'baz' : 10})['bar'] # 12 = 10 + 1
```

This breaks, it should get `11`

### PR DESCRIPTION
**Pull request by elijahbenizzy**
We just weren't passing it to the node-determining function, so raw_execute wasn't calling the validator with the right inputs.

Fixes #307

[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: hamilton/driver.py [hunk 1]
```diff
@@ -581,21 +582,30 @@ def _visualize_execution_helper(
...
-        # _final_vars = self._create_final_vars(final_vars)
-        nodes, user_nodes = fn_graph.get_upstream_nodes(final_vars, inputs)
+        nodes, user_nodes = fn_graph.get_upstream_nodes(final_vars, inputs, overrides)
         Driver.validate_inputs(fn_graph, adapter, user_nodes, inputs, nodes)
         node_modifiers = {fv: {graph.VisualizationNodeModifiers.IS_OUTPUT} for fv in final_vars}
         for user_node in user_nodes:
             if user_node.name not in node_modifiers:
                 node_modifiers[user_node.name] = set()
             node_modifiers[user_node.name].add(graph.VisualizationNodeModifiers.IS_USER_INPUT)
+        all_nodes = nodes | user_nodes
```
Comment (skrawcz) [inline hamilton/driver.py:592]: will this work for all versions of python?

Comment (skrawcz) [reply hamilton/driver.py:592]: otherwise why not use the `.union` ?

**Review by skrawcz — APPROVED** [review]

**Comment by elijahbenizzy** [comment]
See as an example 
image

File: hamilton/driver.py (hunk 1 above)
Comment (elijahbenizzy) [reply hamilton/driver.py:592]: Consistent with how we were doing it in another place, and yeah, this was a python 2 feature

**Pull request merged**